### PR TITLE
[clickhouse/exporter] Update create schema config option

### PR DESCRIPTION
**Pull request by SpencerTorres**
#### DEPENDS ON #33693 

**Description:**

A follow-up to #32282 that changes `create_schema` from `*bool` to `bool`, while also properly using the default config / factory.

**Testing:** 
- Updated tests

### CODE REVIEW

**Comment by dmitryax** [comment]
Please rebase. I'm not sure why it depends on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33693

**Comment by SpencerTorres** [comment]
Rebased. Need to fix the next PR but this one should be good now 👍

**Comment by crobert-1** [comment]
Failing test is unrelated, I've opened #33715 to address the failure.

**Review by crobert-1 — APPROVED** [review]
LGTM. I don't think this requires a changelog, but code owners and provide better input there.

File: .chloggen/clickhouse-update-create-schema-config-option.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []
```
Comment (dmitryax) [inline .chloggen/clickhouse-update-create-schema-config-option.yaml:27]: This is an API breaking change. The end-users aren't affected
```suggestion
change_logs: [api]
```

File: .chloggen/clickhouse-update-create-schema-config-option.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```
Comment (dmitryax) [inline .chloggen/clickhouse-update-create-schema-config-option.yaml:4]:
```suggestion
change_type: breaking
```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**